osu.Game/Online/Rooms/PlaylistExtensions.cs

```diff
@@ -38,7 +40,20 @@ public static IEnumerable<PlaylistItem> GetUpcomingItems(this IEnumerable<Playli
                 : GetUpcomingItems(playlist).First();
         }
 
+        /// <summary>
+        /// Returns the total duration from the <see cref="PlaylistItem"/> in playlist order from the supplied <paramref name="playlist"/>,
+        /// </summary>
         public static string GetTotalDuration(this BindableList<PlaylistItem> playlist) =>
-            playlist.Select(p => p.Beatmap.Length).Sum().Milliseconds().Humanize(minUnit: TimeUnit.Second, maxUnit: TimeUnit.Hour, precision: 2);
+            playlist.Select(p =>
+            {
+                IEnumerable<Mod> modList = [];
+                if (p.RequiredMods.Length > 0)
+                {
+                    var ruleset = p.Beatmap.Ruleset.CreateInstance();
+                    modList = p.RequiredMods.Select(mod => mod.ToMod(ruleset));
+                }
+                double rate = ModUtils.CalculateRateWithMods(modList);
```

Comment:
Created this validation to deal with the problem of instancing ruleset everytime for everymap